plugins/yallah/src/main/java/de/dfki/vsm/xtension/yallah/YallahExecutor.java

```diff
@@ -0,0 +1,301 @@
+                mLogger.message("Remove active activity actions to avoid deadlocks...");
+                synchronized (mActivityWorkerMap) {
+                    mActivityWorkerMap.clear();
+                    // wake me up ..
+                    mActivityWorkerMap.notifyAll();
+                }
+
+            });
+        });
+
+
+        //
+        //
+        String launch_mode = mConfig.getProperty("launchMode") ;
+        YallahProperties.LaunchMode lm = YallahProperties.LaunchMode.valueOf(launch_mode) ;
+        switch (lm) {
+            case None:
+                break ;
+            case App:
+                // TODO -- launch a standalone app
+                mLogger.failure("Launch mode " + lm.toString() + "not supported yet. YALLAH not launched.");
+                break;
+            case WebPage:
+                // TODO -- run an internal web server that provides the webpages and open them in a browser.
+                mLogger.failure("Launch mode " + lm.toString() + "not supported yet. YALLAH not launched.");
+                break;
+            default:
+                mLogger.warning("Unrecognized launch method " + lm.toString() );
+        }
+
+
+        //
+        mLogger.message("YALLAH plugin launched.");
+    }
+
+    @Override
+    public void unload() {
+        mLogger.message("YALLAH plugin unloading....");
+
+        //
+        // TODO -- 1. Stop the YALLAH application (Not much to do if it is a WebPage)
+
+        // 2. Shutdown the WebSocket server
+        mWebSocketServer.stop();
+        mSocketConnectCtx = null ;
+
+        mLogger.message("YALLAH plugin unloaded.");
+    }
+
+
+    @Override
+    public void execute(AbstractActivity activity) {
+
+        mLogger.message("YALLAH Agent '" + activity.getActor() + "' said: " + activity.getText());
+        mLogger.message("Activity name: " + activity.getName() + ", type: " + activity.getType() + ", features: " + activity.getFeatures());
+
+        if (activity instanceof SpeechActivity) {
+            //
+            // SPEECH activity
+            SpeechActivity sa = (SpeechActivity) activity;
+
+            String punct = sa.getPunct() ;
+            String text_only = sa.getTextOnly(MARKER).trim() ;
+            LinkedList<String> time_marks = sa.getTimeMarks(MARKER);
+
+            mLogger.message("This is a Speech Activity. text only: '" + text_only + "'; punct: '" + punct + "'"
+                    + "There are " + time_marks.size() + " time marks") ;
+            time_marks.forEach(mLogger::message);
+
+            if (text_only.isEmpty()) {
+                //
+                // If text is empty, there is no need to send the marker to the client:
+                // execute them all of them immediately.
+
+                LinkedList<String> timemarks = sa.getTimeMarks(YallahExecutor.MARKER);
+                for (String tm : timemarks) {
+                    // mLogger.warning("Directly executing activity at timemark " + tm);
```

Comment:
Commented out logging. Is this from debugging? If so, please remove before release